components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/mediators/MockDBPersistMediator.java

```diff
@@ -0,0 +1,80 @@
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+package org.wso2.carbon.apimgt.gateway.mediators;
+
+import com.google.gson.JsonObject;
+import java.util.concurrent.ConcurrentHashMap;
+import com.google.gson.JsonParser;
+import org.apache.synapse.MessageContext;
+import org.apache.synapse.mediators.AbstractMediator;
+import org.wso2.carbon.apimgt.impl.APIConstants;
+
+public class MockDBPersistMediator extends AbstractMediator {
+    private static final ConcurrentHashMap<String, String> mockDBMap = new ConcurrentHashMap<>();
+    final String API_UUID = "API_UUID";
+    final String MOCK_DB = "mockDB";
+    final String OPEN_API_STRING = "OPEN_API_STRING";
+
+    @Override
+    public boolean mediate(MessageContext context) {
+        try {
+            String openAPIString = (String) context.getProperty(OPEN_API_STRING);
+            JsonObject openAPI = JsonParser.parseString(openAPIString).getAsJsonObject();
+            String apiKey = Integer.toString(
+                    (context.getProperty(API_UUID) +
+                            (openAPI.get(APIConstants.X_WSO2_MOCKDB) != null ?
+                                    openAPI.get(APIConstants.X_WSO2_MOCKDB).getAsString() : "")).hashCode());
+
+            if (context.getProperty(MOCK_DB) == null) { // Only for request
+                if (!mockDBMap.containsKey(apiKey)) {
+                    setMockDB(openAPI, apiKey);
+                    if (log.isDebugEnabled()) {
+                        log.debug("MockDB Initialized: " + mockDBMap.get(apiKey));
+                    }
+                }
+                // Set the mockDB to the context
+                context.setProperty(MOCK_DB, mockDBMap.get(apiKey));
+                if (log.isDebugEnabled()) {
+                    log.debug("MockDB set to Context: " + mockDBMap.get(apiKey));
+                }
+            } else { // Only for response
+                // Update the mockDB for the specific API
+                updateMockDB(apiKey, (String) context.getProperty(MOCK_DB));
+                if (log.isDebugEnabled()) {
+                    log.debug("MockDB Updated: " + mockDBMap.get(apiKey));
+                }
+            }
+            return true; // Successful mediation
+        } catch (Exception e) {
+            log.error("Error in MockDBPersistMediator: " + e.getMessage());
+            e.printStackTrace();
+            return false;
```

Comment:
If there is an error during the mock DB persistence here, how would it affect the API invocation?